### ISSUE TITLE
Parse error when using short arrow functions as default value in destructured object function argument. 

### ISSUE THREAD

**Issue by WRidder** [eslint/eslint#4622]
I have the following construct:

```javascript
// Function accepts an object with an action property. 
// If the property is not set, set a default arrow function as value
// If no object is passed at all, use an empty object as initial value as function argument.
const someVar = ({actions = () => ({})} = {}) => {

};
```

This raises an error when running eslint: `parsing error: unexpected token =>`

However, the following does not raise an error:
```javascript
const defaultFunction = () => ({});
const someVar = ({actions = defaultFunction)} = {}) => {

};
```

Both versions do work correctly when executed btw.

My .eslintrc config:
```javascript
{
  "extends": "airbnb",
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "globals": {
    "React": true
  },
  "rules": {
    "valid-jsdoc": 1,
    "quotes": [
      2,
      "double"
    ],
    "new-cap": 0,
    "brace-style": [2, "stroustrup", { "allowSingleLine": false }],
    "id-length": [2, {"min": 1, "properties": "never"}],
    "no-param-reassign": 0,

    // Disable until Flow supports let and const
    "no-var": 0,
    "vars-on-top": 0,

    // Disable comma-dangle unless need to support it
    "comma-dangle": 0
  },
  "root": true
}
```

Running on eslint versions:
```
    "eslint": "1.9.0",
    "eslint-config-airbnb": "1.0.0",
    "eslint-plugin-react": "3.9.0",
```
`

**Comment by eslintbot** [comment]
@WRidder Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
This is an Espree issue, can you file an issue there? https://github.com/eslint/espree/issues
